contrib/src/main/scala/akka/stream/contrib/AccumulateWhileUnchanged.scala

```diff
@@ -63,19 +83,22 @@ final class AccumulateWhileUnchanged[Element, Property](propertyExtractor: Eleme
 
         currentState match {
           case Some(`nextState`) =>
-            buffer += nextElement
-            pull(in)
+            if (maxElements.isDefined && nbElements >= maxElements.get) {
```

Comment:
`maxElements` is an option, so this `if/else` can be rewritten to:

```scala
maxElements match {
  case Some(max) if nbElements >= max => pushResults(...)
  case _ =>
    ...
}
```
Or, even better, it can be flattened to the enclosing match of the `currentState`.

Reply:
The best I can think of with one match is:

```
(currentState, maxElements) match {
  case (Some(`nextState`), max) if max.isEmpty || nbElements < max.get =>
    buffer += nextElement
    nbElements += 1
    pull(in)
  case (_, _) =>
    pushResults(Some(nextElement), Some(nextState))
}
```

What do you think?